### ISSUE TITLE
The pyright CLI ignores scripts without a `.py` extension

### ISSUE THREAD

**Issue by fwcd** [microsoft/pyright#3398]
**Describe the bug**
Passing a Python script without a `.py` extension to the `pyright` CLI results in an (incorrect?) file-not-found message.

**To Reproduce**
Create a script, e.g. in `./test-script`:

```python
#!/usr/bin/env python3

def f(x: int):
    pass

f("abc")
```

Running `pyright test-script` now yields:

```
No configuration file found.
No pyproject.toml file found.
stubPath .../typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
File or directory ".../test-script" does not exist.
No source files found.
Completed in 0.289sec
```

**Expected behavior**
The same result that `pyright test-script.py` yields when I rename the script:

```
No configuration file found.
No pyproject.toml file found.
stubPath .../typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
.../test-script.py
  .../test-script.py:6:3 - error: Argument of type "Literal['abc']" cannot be assigned to parameter "x" of type "int" in function "f"
    "Literal['abc']" is incompatible with "int" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.435sec
```

**VS Code extension or command-line**
Command-line tool, pyright 1.1.241

**Comment by erictraut** [comment]
Yes, this is by design. Pyright operates only on ".py" and ".pyi" files. If you rename your script to end in ".py", it will work.

**Comment by erictraut** [comment]
Yeah, I agree the error message could be clearer. I've made the suggested improvement, and it will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published.

**Comment by bulletmark** [comment]
Sorry, but this design decision seems very odd. A command line program does not normally have a `.py` extension and indeed it looks ugly if it does. The language a CLI program is written in is completely irrelevant to a user and users also don't want to have to type `.py`. 

If I create a command line program, say called `myprog`, then I can't use pyright:

```
$ pyright myprog
File "/home/bulletmark/myprog" is not a Python source file.
```

I just can't see any good reason why this constraint is necessary for **directly specified files**, and I urge you to please re-open this issue.

**Comment by bulletmark** [comment]
I see this bug was fixed in version 1.1.317. Thanks.
